### PR TITLE
add support for Brueckner...

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: experimental/GModule/Cohomology.jl [hunk 1]
```diff
@@ -749,6 +804,10 @@ function cohomology_group(C::GModule{PermGroup,GrpAbFinGen}, i::Int)
   end
   error("only H^0, H^1 and H^2 are supported")
 end
+#TODO:
+#     versions of fp_group(GrpAbFinGen) and extension
+#     to return PcGroups
+#     (via 1st argument being PcGroup)
```
Comment (fingolfin) [inline experimental/GModule/Cohomology.jl]: So a way to convert a (finite) GrpAbFinGen into a PcGroup -- we can add that (CC @ThomasBreuer )

File: experimental/GModule/Cohomology.jl [hunk 2]
```diff
@@ -763,11 +822,17 @@ function fp_group(M::GrpAbFinGen)
...
+
 """
 Given a 2-cocycle, return the corresponding group extension, ie. the large
-group, the injection of the abelian group and the quotient.
+group, the injection of the abelian group and the quotient as well as a map
+that given a tuple of elements in the group and the abelian group returns
+the corresponding elt in the extension. 
 """
-function extension(c::CoChain{2,PermGroupElem})
+function extension(c::CoChain{2,<:Oscar.GAPGroupElem})
```
Comment (fingolfin) [inline experimental/GModule/Cohomology.jl:1254]: TODO: we need a version for this that takes PcGroups and produces PcGroups

File: experimental/GModule/GModule.jl [hunk 3]
```diff
@@ -536,7 +541,7 @@ end
 if isdefined(Hecke, :stub_composition_factors)
   function Hecke.stub_composition_factors(x::Vector{T}) where {T}
     F = base_ring(x[1])
-    h = Oscar.iso_oscar_gap(F)
+    h = Oscar.ring_iso_oscar_gap(F)
```
Comment (fingolfin) [inline experimental/GModule/GModule.jl]: We renamed Oscar.ring_iso_oscar_gap to Oscar.iso_oscar_gap ...

**Review by fingolfin — CHANGES_REQUESTED** [review]
- `ring_iso_oscar_gap` must be changed to `iso_oscar_gap`
- many lines are not covered by tests, hence the above issue is not detect by the tests. I don't want to block the PR over this, but I think it'd be good to have a few (even trivial!) tests otherwise we'll just break this code constantly

**Comment by fieker** [comment]
On Wed, Jan 19, 2022 at 09:30:59AM -0800, Max Horn wrote:
> @fingolfin requested changes on this pull request.
> 
> - `ring_iso_oscar_gap` must be changed to `iso_oscar_gap`
> - many lines are not covered by tests, hence the above issue is not detect by the tests. I don't want to block the PR over this, but I think it'd be good to have a few (even trivial!) tests otherwise we'll just break this code constantly
> 
> 
> 
in PR 970 are some tests

File: experimental/GModule/Cohomology.jl (hunk 1 above)
Comment (fieker) [reply experimental/GModule/Cohomology.jl]: done

File: experimental/GModule/Cohomology.jl (hunk 2 above)
Comment (fieker) [reply experimental/GModule/Cohomology.jl:1254]: done

File: experimental/GModule/GModule.jl (hunk 3 above)
Comment (fieker) [reply experimental/GModule/GModule.jl]: just done, but still:
```
julia> Oscar.RepPc.reps(abelian_closure(QQ)[1], G)
ERROR: failed to convert GAP.GapObj to any known type:
 GAP: !Z(137)^0
Stacktrace:
  [1] gap_to_julia(x::GAP.GapObj; recursive::Bool)
    @ GAP ~/.julia/dev/GAP/src/gap_to_julia.jl:321
  [2] #gap_to_julia#29
    @ ~/.julia/dev/GAP/src/gap_to_julia.jl:94 [inlined]
  [3] #gap_to_julia#27
    @ ~/.julia/dev/GAP/src/gap_to_julia.jl:89 [inlined]
  [4] #36
    @ ~/.julia/dev/GAP/src/gap_to_julia.jl:204 [inlined]
```
I thought this was solved?

**Review by fingolfin — APPROVED** [review]

**Pull request merged**